src/sicks3000.cc

```diff
@@ -310,3 +310,330 @@ unsigned short SickS3000::CreateCRC(uint8_t *Data, ssize_t length)
+  SEVEN_SEGMENT_CODE(SEG_L,SEG_9), //There is a short-circuit between the reset input and another input or output, or the reset pulse does not comply with the requirements. (L. - 9.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG__1), //Input signal for an undefined monitoring case (n. - _1.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_2), //Incorrect sequence on switching the monitoring cases (n. - 2.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_3), //Incorrect operation of the control inputs (n. - 3.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_4), //Short-circuit on control inputs A1/2 or incorrect operation of A1/2 via EFI (n. - 4.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_5), //Short-circuit on control inputs B1/2 or incorrect operation of B1/2 via EFI (n. - 5.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_6), //Short-circuit on control inputs C1/2 or incorrect operation of C1/2 via EFI (n. - 6.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_7), //Short-circuit on control inputs D1/2 or incorrect operation of D1/2 via EFI (n. - 7.)
+  SEVEN_SEGMENT_CODE(SEG_n,SEG_8), //Incorrect operation of E1/2 via EFI (n. - 8.)
+  SEVEN_SEGMENT_CODE(SEG_o,SEG_0), //Park/stand-by mode (see section 4.10.7 on page 64); the OSSDs are in the OFF state; the laser is deactivated. (o.)
+  SEVEN_SEGMENT_CODE(SEG_P,SEG_0), //A device connected via EFI reports a malfunction. (P.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_u), //Front screen calibration active (u. flashing)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG__1), //Channel 1 of the contamination measurement contaminated, i.e. the front screen is dirty (u. - _1.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_2), //Channel 2 of the contamination measurement contaminated, i.e. the front screen is dirty (u. - 2.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_3), //Channel 2 of the contamination measurement contaminated, i.e. the front screen is dirty (u. - 3.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_4), //Channel 2 of the contamination measurement contaminated, i.e. the front screen is dirty (u. - 4.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_5), //Channel 2 of the contamination measurement contaminated, i.e. the front screen is dirty (u. - 5.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_6), //Channel 2 of the contamination measurement contaminated, i.e. the front screen is dirty (u. - 6.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_7), //No front screen fitted or dazzling of the contamination measurement (u. - 7.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_8), //Traceability data incorrect or front screen calibration failed (u. - 8.)
+  SEVEN_SEGMENT_CODE(SEG_u,SEG_9), // *see above (u. - 9.)
+  SEVEN_SEGMENT_CODE(SEG_y,SEG__1), //Internal error in the S3000 (y. - _1.)
+  SEVEN_SEGMENT_CODE(SEG_y,SEG_2), //Internal I/O module error(y. - 2.)
+  SEVEN_SEGMENT_CODE(SEG_y,SEG_3), //I/O module/sensor head device combination invalid (y. - 3.)
+};
+
+const char* s3000_error_causes[NUM_S3000_ERROR_CODES] = {
+  
+  //Indication of protective field and contour infringements in the dual field mode
+  "Object in protective field (1_.)",
```

Comment:
This seems incredibly error-prone, separating the strings from the codes. What was the reason not to use an array of structs?